headers/addons/tilt.h

```diff
@@ -0,0 +1,118 @@
+#ifndef _Tilt_H
+#define _Tilt_H
+
+#include "gpaddon.h"
+
+#include "GamepadEnums.h"
+
+#ifndef TILT_ENABLED
+#define TILT_ENABLED 1
+#endif
+
+#ifndef PIN_TILT_1
+#define PIN_TILT_1 -1
+#endif
+
+#ifndef PIN_TILT_2
+#define PIN_TILT_2 -1
+#endif
+
+#ifndef PIN_TILT_FUNCTION
+#define PIN_TILT_FUNCTION -1
+#endif
+
+#ifndef PIN_TILT_LEFT_ANALOG_UP
+#define PIN_TILT_LEFT_ANALOG_UP -1
+#endif
+
+#ifndef PIN_TILT_LEFT_ANALOG_DOWN
+#define PIN_TILT_LEFT_ANALOG_DOWN -1
+#endif
+
+#ifndef PIN_TILT_LEFT_ANALOG_LEFT
+#define PIN_TILT_LEFT_ANALOG_LEFT -1
+#endif
+
+#ifndef PIN_TILT_LEFT_ANALOG_RIGHT
+#define PIN_TILT_LEFT_ANALOG_RIGHT -1
+#endif
+
+#ifndef PIN_TILT_RIGHT_ANALOG_UP
+#define PIN_TILT_RIGHT_ANALOG_UP -1
+#endif
+
+#ifndef PIN_TILT_RIGHT_ANALOG_DOWN
+#define PIN_TILT_RIGHT_ANALOG_DOWN -1
+#endif
+
+#ifndef PIN_TILT_RIGHT_ANALOG_LEFT
+#define PIN_TILT_RIGHT_ANALOG_LEFT -1
+#endif
+
+#ifndef PIN_TILT_RIGHT_ANALOG_RIGHT
+#define PIN_TILT_RIGHT_ANALOG_RIGHT -1
+#endif
+
+#ifndef TILT_SOCD_MODE
+#define TILT_SOCD_MODE SOCD_MODE_NEUTRAL
+#endif
+
+// Tilt Module Name
+#define TiltName "Tilt"
+
+// These constants define the adjustment factors for gamepad analog inputs under different tilt states.
+// Defines the behavior of the left analog stick when the Tilt1 and Tilt2 buttons are pressed.
+// The main purpose of the left analog stick is to move the character.
+// Pressing it simultaneously with Tilt 1 will make the character walk.
+// Pressing it simultaneously with Tilt 2 will make the character walk more slowly.
+#define TILT1_FACTOR_LEFT_X 0.35  // Adjustment factor for left analog stick X direction
```

Comment:
I'm not too familiar with the use case for the Tilt function. Should the values defined here for `TILT#_FACTOR...` be adjustable by a user via the web config?

Reply:
I do not believe that I currently need to be able to change them in the webconfig. I have moved those factors to tilt.cpp. I originally just moved them out of the function to make the code easier to read.